channels/applicationinsights-channel-js/src/Sender.ts

```diff
@@ -268,12 +268,13 @@ export class Sender extends BaseTelemetryPlugin implements IChannelControls {
                     if (config.storagePrefix){
                         utlSetStoragePrefix(config.storagePrefix);
                     }
-                    if (config.crossOriginResourcePolicy){
-                        this.addHeader(CrossOriginResourcePolicyHeader, config.crossOriginResourcePolicy);
-                    }
+                   
                     let ctx = createProcessTelemetryContext(null, config, core);
                     // getExtCfg only finds undefined values from core
                     let senderConfig = ctx.getExtCfg(identifier, defaultAppInsightsChannelConfig);
+                    if (senderConfig.corsPolicy){
+                        this.addHeader(CrossOriginResourcePolicyHeader, senderConfig.corsPolicy);
```

Comment:
As this is inside of an onConfigChange(), this will get called (potentially) more than once so we will need to
- protect or handling setting the same header more than once (which should already be handled by the way addHeader works)
- Removing this header if the corsPolicy is removed afterwards (unlikely, but not impossible)